downloads/swan-lake-release-notes/swan-lake-2201.3.0/RELEASE_NOTE.md

```diff
@@ -0,0 +1,468 @@
+
+```ballerina
+type User record {|
+    readonly int id;
+    string name;
+|};
+
+type Login record {|
+    int userId;
+    string time;
+|};
+
+type Entry record {|
+    string? name;
+    string login;
+|};
+
+public function main() {
+    Login[] logins = [
+        {userId: 6789, time: "20:10:23"},
+        {userId: 1234, time: "10:30:02"},
+        {userId: 3987, time: "12:05:00"}
+    ];
+
+    table<User> key(id) users = table [
+            {id: 1234, name: "Keith"},
+            {id: 6789, name: "Anne"}
+        ];
+
+    Entry[] entities = from var login in logins
+        outer join var user in users
+        on login.userId equals user?.id
+        select {name: user?.name, login: login.time};
+
+    // Now, the code below gives an error at the typed binding pattern of the outer join clause.
+    entities = from var login in logins
+        outer join User user in users  // error: outer join must be declared with 'var'
+        on login.userId equals user.id
+        select {name: user?.name, login: login.time};
+}
+```
+
+#### Disallow method calls with record fields of function types
+
+Previously, method calls were allowed with record fields of function types similar to object fields. This has now been disallowed since it is not allowed by the language specification.
+
+```ballerina
+type Foo record {|
+    function (int a, int b) returns int func;
+|};
+
+function callFunction(Foo foo) {
+    int _ = foo.func(1, 2); // error
+}
+```
+
+## Compiler API updates
+
+### New features
+
+#### Semantic API
+Added a new `annotAttachments()` API to get the annotation attachments and their constant values from the annotatable symbols
+
+### Improvements
+
+#### Semantic API
+Improved the `constValue()` method to retrieve the constant value as an object from the constant symbol
+
+#### Syntax API
+
+Added a few methods to the `NodeParser` API. The methods below have been introduced for the `NodeParser` class.
+  - `Node parseObjectMember(String text)`
+  - `ModulePartNode parseModulePart(String text)`
+  - `IntermediateClauseNode parseIntermediateClause(String text, boolean allowActions)`
+
+### Backward-incompatible changes
+
+Updated the `NodeFactory` methods to allow optional terminating semicolons for module-level declarations. The methods below in the `NodeFactory` have been updated with an extra parameter for the optional semicolon token.
+
+  - `createServiceDeclarationNode`
+  - `createFunctionBodyBlockNode`
+  - `createEnumDeclarationNode`
+  - `createClassDefinitionNode`
+
+### Bug fixes
+
+To view bug fixes, see the [GitHub milestone for 2201.3.0 (Swan Lake)](https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+milestone%3A2201.3.0+label%3AType%2FBug+is%3Aclosed+label%3ATeam%2FCompilerFETools+).
+
+## Runtime updates
+
+### Improvements
+
+#### Strand dump showing the total strand count
```

Comment:
Shouldn't we rename this to `Show the total strand count in strand dump` for consistency? @praneesha / @warunalakshitha ?

Reply:
Changed.